tests/test_utils/test_transactiontestcase.py

```diff
@@ -26,3 +45,22 @@ def test(self, call_command):
             reset_sequences=False, inhibit_post_migrate=True,
             database='default', verbosity=0,
         )
+
+
+class TestDataConsistencyOnTearDown(TestSerializedContentMockMixin, TransactionTestCase):
+    """
+    TransactionTestCase._fixture_teardown() is flushing the whole database.
+    If an other test after this one needs some data coming from initial data migrations,
+    there data won't be found.
+    One solution is to load again initial data migrations after the flush called on _fixture_teardown().
+    """
+    available_apps = ['test_utils']
+    serialized_rollback = True
+    initial_data_migration = '[{"model": "test_utils.car", "pk": 1, "fields": {"name": "K 2000"}}]'
+
+    def _post_teardown(self):
+        super(TestDataConsistencyOnTearDown, self)._post_teardown()
+        self.assertTrue(Car.objects.exists())
```

Comment:
The `initial_data_migration` mock represents which data have been serialised by Django and coming from initial data migration(s).

I want to emulate this `Car` object to be created from these initial data migrations.

It should still exists at the end of the `TransactionTestCase`, that's the idea behind this MR. If another test after this one needs this initial data migration, it will still be there.  If there is no other test after this one, then the database will be in the same state as after the initial data migration(s) (specially important if we use `--keepdb` option).

I have tried to summarise the current situation: http://5minutes.youkidea.com/transactiontestcase-keepdb-django-issues.html